lib/s3routes/routes.ts

```diff
@@ -221,17 +235,18 @@ export default function routes(
 
     if (!req.url!.startsWith('/_/healthcheck') &&
         !req.url!.startsWith('/_/report')) {
-        log.info('received request', clientInfo);
+        log.debug('received request', clientInfo);
     }
 
-    log.end().addDefaultFields(clientInfo);
+    log.addDefaultFields(clientInfo);
```

Comment:
removing `end()` means we may not see any log until the request is complete: which may make it harder to troubleshoot requests which somehow do not terminate. Is this really what we want?

Reply:
The `end` function in werelogs should only be used to end the RequestLogger. It returns an `EndLogger` that should only be used to log one last time. It also allows to have the "elapsed time" info. But as long as we log using the current logging level, the logs will be flushed in real time. When the dump level is reached, we might flush those logs again, but with more info (like, debug logs).

I am not sure to understand what you mean by "we may not see any log until the request is complete"?

Reply:
> as long as we log using the current logging level, the logs will be flushed in real time

this answers the "we may not see any log until the request is complete" issue, so no problem actually :-)